cpp/src/codegen/tree_expr_builder.cc

```diff
@@ -65,6 +73,9 @@ NodePtr TreeExprBuilder::MakeNull(DataTypePtr data_type) {
     return std::make_shared<LiteralNode>(data_type, LiteralHolder((float_t)0), true);
   case arrow::Type::DOUBLE:
     return std::make_shared<LiteralNode>(data_type, LiteralHolder((double_t)0), true);
+  case arrow::Type::STRING:
+  case arrow::Type::BINARY:
+    return std::make_shared<LiteralNode>(data_type, LiteralHolder(std::string("")), true);
```

Comment:
[for my understanding] - does this lead to a new string every time we want string or binary null? can this be static since all we need is the flag?

Reply:
yes, will fix